### PR TITLE
Merge upgrade jobs

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Changes proposed in this pull request:

- Merge the two upgrade jobs to have one job that executes the E2E upgrade test scenarios and the external solution integration tests before and after upgrading Kyma.
- Remove the optional upgrade job as it is not needed any more.

**Related issue(s)**

Fixes https://github.com/kyma-project/kyma/issues/8799

### CODE REVIEW

File: prow/scripts/cluster-integration/kyma-gke-upgrade.sh [hunk 1]
```diff
@@ -344,40 +346,59 @@ function checkTestPodTerminated() {
...
-    set -o errexit
-
-    echo "Logs for prepare data operation to test e2e upgrade: "
-    # shellcheck disable=SC2046
-    kubectl logs -n "${UPGRADE_TEST_NAMESPACE}" $(kubectl get pod -n "${UPGRADE_TEST_NAMESPACE}" -l "${UPGRADE_TEST_RESOURCE_LABEL}=${UPGRADE_TEST_LABEL_VALUE_PREPARE}" -o json | jq -r '.items | .[] | .metadata.name') -c "${TEST_CONTAINER_NAME}"
-    if [ "${prepareTestResult}" != 0 ]; then
-        echo "Exit status for prepare upgrade e2e tests: ${prepareTestResult}"
-        exit "${prepareTestResult}"
-    fi
+    installTestChartOrFail "${UPGRADE_TEST_PATH}" "${UPGRADE_TEST_RELEASE_NAME}" "${UPGRADE_TEST_NAMESPACE}"
```
Comment (nachtmaar) [inline prow/scripts/cluster-integration/kyma-gke-upgrade.sh:398]: Maybe that comment makes it a bit easier to read 😎 
```suggestion
    # install upgrade test
    installTestChartOrFail "${UPGRADE_TEST_PATH}" "${UPGRADE_TEST_RELEASE_NAME}" "${UPGRADE_TEST_NAMESPACE}"
```

File: prow/scripts/cluster-integration/kyma-gke-upgrade.sh [hunk 2]
```diff
@@ -344,40 +346,59 @@ function checkTestPodTerminated() {
...
+    installTestChartOrFail "${EXTERNAL_SOLUTION_TEST_PATH}" "${EXTERNAL_SOLUTION_TEST_RELEASE_NAME}" "${EXTERNAL_SOLUTION_TEST_NAMESPACE}"
```
Comment (nachtmaar) [inline prow/scripts/cluster-integration/kyma-gke-upgrade.sh:399]: Maybe that comment makes it a bit easier to read 😎 

```suggestion
    # install external-solution test
    installTestChartOrFail "${EXTERNAL_SOLUTION_TEST_PATH}" "${EXTERNAL_SOLUTION_TEST_RELEASE_NAME}" "${EXTERNAL_SOLUTION_TEST_NAMESPACE}"
```

File: prow/scripts/cluster-integration/kyma-gke-upgrade.sh [hunk 3]
```diff
@@ -344,40 +346,59 @@ function checkTestPodTerminated() {
     return 1
 }
 
-createTestResources() {
-    shout "Create e2e upgrade test resources"
-    date
+function installTestChartOrFail() {
```
Comment (sayanh) [inline prow/scripts/cluster-integration/kyma-gke-upgrade.sh:349]: Would it make sense to move this to the `library.sh"`?

File: prow/scripts/cluster-integration/kyma-gke-upgrade.sh [hunk 4]
```diff
@@ -344,40 +346,59 @@ function checkTestPodTerminated() {
...
-    injectTestingAddons
+  shout "Create ${name} resources"
+  date
 
-    if [  -f "$(helm home)/ca.pem" ]; then
-        local HELM_ARGS="--tls"
-    fi
+  local HELM_ARGS
+  if [[ -f "$(helm home)/ca.pem" ]]; then
+      HELM_ARGS="--tls"
```
Comment (sayanh) [inline prow/scripts/cluster-integration/kyma-gke-upgrade.sh:359]: Is migration to Helm3 a really simple task that can be done here? Just a thought only. I won't block the PR.

File: prow/scripts/cluster-integration/kyma-gke-upgrade.sh [hunk 5]
```diff
@@ -344,40 +346,59 @@ function checkTestPodTerminated() {
...
+      exit "${prepareResult}"
+  fi
+}
 
-    helm install "${UPGRADE_TEST_PATH}" \
-        --name "${UPGRADE_TEST_RELEASE_NAME}" \
-        --namespace "${UPGRADE_TEST_NAMESPACE}" \
-        --timeout "${UPGRADE_TEST_HELM_TIMEOUT_SEC}" \
-        --wait ${HELM_ARGS}
+function waitForTestPodToFinish() {
```
Comment (sayanh) [inline prow/scripts/cluster-integration/kyma-gke-upgrade.sh:376]: Another candidate for `library.sh`, I guess or can be scooped out to be an one-thing-do-well script inside `test-infra/prow/scripts`

File: prow/scripts/cluster-integration/kyma-gke-upgrade.sh (hunk 3 above)
Comment (marcobebway) [reply prow/scripts/cluster-integration/kyma-gke-upgrade.sh:349]: As discussed offline, we will keep it like that for now.

File: prow/scripts/cluster-integration/kyma-gke-upgrade.sh (hunk 4 above)
Comment (marcobebway) [reply prow/scripts/cluster-integration/kyma-gke-upgrade.sh:359]: As discussed offline, we will keep it like that for now.

File: prow/scripts/cluster-integration/kyma-gke-upgrade.sh (hunk 5 above)
Comment (marcobebway) [reply prow/scripts/cluster-integration/kyma-gke-upgrade.sh:376]: As discussed offline, we will keep it like that for now.

File: prow/scripts/cluster-integration/kyma-gke-upgrade.sh (hunk 4 above)
Comment (sayanh) [reply prow/scripts/cluster-integration/kyma-gke-upgrade.sh:359]: Not applicable until we release 1.15

File: prow/scripts/cluster-integration/kyma-gke-upgrade.sh (hunk 1 above)
Comment (marcobebway) [reply prow/scripts/cluster-integration/kyma-gke-upgrade.sh:399]: Done

File: prow/scripts/cluster-integration/kyma-gke-upgrade.sh (hunk 2 above)
Comment (marcobebway) [reply prow/scripts/cluster-integration/kyma-gke-upgrade.sh:402]: Done

**Pull request merged**